plugins/database/redshift/redshift.go

```diff
@@ -229,132 +172,180 @@ func (r *RedShift) CreateUser(ctx context.Context, statements dbplugin.Statement
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	renewStmts := statements.Renewal
-	if len(renewStmts) == 0 {
-		renewStmts = []string{defaultRenewSQL}
-	}
-
 	db, err := r.getConnection(ctx)
 	if err != nil {
-		return err
+		return dbplugin.UpdateUserResponse{}, err
 	}
 	defer db.Close()
 
 	tx, err := db.BeginTx(ctx, nil)
 	if err != nil {
-		return err
+		return dbplugin.UpdateUserResponse{}, err
 	}
 	defer func() {
 		tx.Rollback()
 	}()
 
-	expirationStr, err := r.GenerateExpiration(expiration)
-	if err != nil {
-		return err
-	}
+	if req.Expiration != nil {
+		renewStmts := req.Expiration.Statements
+		if len(renewStmts.Commands) == 0 {
+			renewStmts.Commands = []string{defaultRenewSQL}
+		}
 
-	for _, stmt := range renewStmts {
-		for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
-			query = strings.TrimSpace(query)
-			if len(query) == 0 {
-				continue
+		expirationStr := req.Expiration.NewExpiration.UTC().Format("2006-01-02 15:04:05")
+
+		for _, stmt := range renewStmts.Commands {
+			for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+				query = strings.TrimSpace(query)
+				if len(query) == 0 {
+					continue
+				}
+
+				m := map[string]string{
+					"name":       req.Username,
+					"expiration": expirationStr,
+				}
+				if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+					return dbplugin.UpdateUserResponse{}, err
+				}
 			}
+		}
+	}
 
-			m := map[string]string{
-				"name":       username,
-				"expiration": expirationStr,
-			}
-			if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
-				return err
+	if req.Password != nil {
+		username := req.Username
+		password := req.Password.NewPassword
+		if username == "" || password == "" {
+			return dbplugin.UpdateUserResponse{}, errors.New("must provide both username and password")
+		}
+
+		// Check if the role exists
+		var exists bool
+		err = db.QueryRowContext(ctx, "SELECT exists (SELECT usename FROM pg_user WHERE usename=$1);", username).Scan(&exists)
+		if err != nil && err != sql.ErrNoRows {
+			return dbplugin.UpdateUserResponse{}, err
+		}
+
+		// Vault requires the database user already exist, and that the credentials
+		// used to execute the rotation statements has sufficient privileges.
+		statements := req.Password.Statements.Commands
+		if len(statements) == 0 {
+			statements = []string{defaultRotateRootCredentialsSQL}
+		}
+		// Execute each query
+		for _, stmt := range statements {
+			for _, query := range strutil.ParseArbitraryStringSlice(stmt, ";") {
+				query = strings.TrimSpace(query)
+				if len(query) == 0 {
+					continue
+				}
+
+				m := map[string]string{
+					"name":     username,
+					"username": username,
+					"password": password,
+				}
+				if err := dbtxn.ExecuteTxQuery(ctx, tx, m, query); err != nil {
+					return dbplugin.UpdateUserResponse{}, err
+				}
 			}
 		}
 	}
 
-	return tx.Commit()
+	return dbplugin.UpdateUserResponse{}, tx.Commit()
```

Comment:
nit: I'm not a fan of putting calls during a return unless that's the entirety of the return statement. I find it makes it harder to follow.